src/screens/Classroom/ClassroomHome.js

```diff
@@ -33,39 +35,68 @@ const styles = StyleSheet.create({
   },
 });
 
-/* const post1 = {
-  author: 'Timmy Turner',
-  text: "I wish that my code doesn't break I really want some pie
-  right now I wonder how people are gopnna see what im typing pls
-  bro just i want to eat i rlly want food bro pls",
-  time: '10:16 PM',
-};
+// eslint-disable-next-line no-unused-vars
+export default function ClassroomHome({ navigation }) {
+  const [initializing, setInitializing] = useState(true);
+  const uid = navigation.getParam('uid', null);
+  const classroomInfo = navigation.getParam('classroomInfo', null);
+  const code = navigation.getParam('code', null);
+  const ref = Firestore().collection('users');
+  const [userState, setUserState] = useState(INITIAL_USER_STATE);
 
-const cl = {
-  classroom: {
-    title: 'CL1',
-    posts: [post1],
-  },
-}; */
+  // const [showLoading, setShowLoading] = useState(false);
+
+  async function getUserData() {
```

Comment:
Let's use a promise chain to retrieve data from Firestore: 

Firestore().collection('users')
.doc(uid)
.get()
.then((document) => {
  if(document.exists) {
     return document.data();
}
  return null;
})
.then((data) => {
  setUserState(data);
})
.catch((e) => {
  Alert.alert(e.message);
});

_This is the same asynchronous code but written with promise chaining._

You can learn more about Promise Chaining here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises

Promises are another type of asynchronous programming supported by JavaScript. The _return_value_ of each stage is passed into the next `.then((_return_value_))`.